### PR TITLE
Modernise witch.js

### PR DESCRIPTION
**Pull request by ConnorKirk**
*Issue #, if available:*

*Description of changes:*
Modernise `witch.js`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/witch/witch.js [hunk 1]
```diff
@@ -10,86 +12,97 @@ const FAILED = "FAILED";
...
-			filelist = walkSync(path.join(dir, file), filelist);
-		} else {
-			filelist.push(path.join(dir, file));
-		}
-	});
-
-	return filelist;
-};
+  const files = fs.readdirSync(dir);
+  filelist = filelist || [];
```
Comment (svozza) [inline source/witch/witch.js:48]: You can get rid of this line by using default paramters, just change the function definition to this:

function walkSync(dir, filelist = []) {

File: source/witch/witch.js [hunk 2]
```diff
@@ -10,86 +12,97 @@ const FAILED = "FAILED";
 
 const BUCKET = process.env.BUCKET;
```
Comment (svozza) [inline source/witch/witch.js:13]: This can be written:

```js
const {BUCKET} = process.env;
```

File: source/witch/witch.js (hunk 1 above)
Comment (ConnorKirk) [reply source/witch/witch.js:48]: Good suggestion!

File: source/witch/witch.js (hunk 2 above)
Comment (ConnorKirk) [reply source/witch/witch.js:13]: Another good spot

**Review by svozza — APPROVED** [review]

**Pull request merged**